ledger/src/blockstore_db.rs

```diff
@@ -178,6 +178,11 @@ pub mod columns {
     #[derive(Debug)]
     /// The root column
     ///
+    /// This column family persists whether a slot has become a root.
+    /// Note that in case there is no entry found for a slot that is older than
+    /// the most recent purged slot, it means this slot has not yet becomed a
+    /// root.
+    ///
```

Comment:
I think you have the logic backwards; **older** than purged slot would mean it could have been a root but we don't know because we purged it. **Newer** than purged would indicate that it could become a root. Also, some of the negations make this a little trick to follow. What do you think about:

"This column family persists whether a slot is a root. Slots on the main fork will be inserted into this column when they are finalized."

This is a little more succinct and I think captures the two points that you were trying to get across:
- Slots could become rooted, the finalized bit captures that
- Slots older than purge slot won't have an entry (this is true for all slot based columns so I think this is not the place to specify that)

Reply:
Oh, yep, yep, should be newer instead xD.

Thanks for suggesting the extra information.  Will include those as well!

Reply:
> Slots older than purge slot won't have an entry (this is true for all slot based columns so I think this is not the place to specify that)

The thing I want to mention in the last part is that NOT_FOUND means the slot is not a root, but I also need to mention the purge as an exception.

Reply:
> but I also need to mention the purge as an exception.

Sure, but I'd argue the note about not having entries older than then where we've purged could be applicable to every slot-based column. Not sure the comment for this single comment is the place to note that

> it means this slot has not yet becomed a root.

Also, I feel like this wording ("not yet") implies every slot will become a root which is not the case